README.md

```diff
@@ -107,8 +107,13 @@ cd agent
 mkdir /tmp/buildkite-builds
 
 # Build an agent binary and start the agent.
-go build -o /usr/local/bin/buildkite-agent .
-buildkite-agent start --debug --build-path=/tmp/buildkite-builds --token "abc"
+go build -o /tmp/buildkite-builds/ .
+
+# Check the version of the build 
+/tmp/buildkite-builds/agent --version
+
+# To run the agent locally
+/tmp/buildkite-builds/agent start --debug --build-path=/tmp/buildkite-builds --token "abc"  --endpoint http://agent.buildkite.localhost/v3
```

Comment:
Ohhhh I see where you were coming from. The README (without this change) was suggesting creating a temp folder (`/tmp/buildkite-builds`) to use as the working directory for the jobs the agent is running, not to build the agent in that folder. `go build -o /usr/local/bin/buildkite-agent .` would build the agent into `/usr/local/bin/`, which is on $PATH by default, meaning when you then run `buildkite-agent start` it uses the one you've built locally. So the instructions there would work unless you already have a `buildkite-agent` in your path (i.e. because you installed it using homebrew etc)

Reply:
ah lol mystery solved. Thanks @dabarrell

Reply:
I'd suggest leaving it closer to the original (not building the binary into a temp folder), but making it not clash with an install of the published version. Maybe @moskyb , @DrJosh9000 or @triarius could suggest the workflow they use

Reply:
Could do something like

```
go build -o /usr/local/bin/buildkite-agent-local .
buildkite-agent-local start --debug --build-path=/tmp/buildkite-builds --token "abc"
```

so you could run both the local and published versions as needed (`buildkite-agent` vs `buildkite-agent-local`), perhaps?

Reply:
@dabarrell I dropped the commit for the README, I think my changes are unnecessary and strictly speaking they don't belong in this PR :) But thanks for all your comments